data_pipeline/gmail_data/get_gmail_data.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+from typing import Iterable
+import pandas as pd
+import backoff
+
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.discovery import Resource
+
+
+def get_gmail_service_for_user_id(secret_file: str, scopes: str, user_id: str) -> Resource:
+    credentials = service_account.Credentials.from_service_account_file(
+        secret_file, scopes=scopes)
+    delegated_credentials = credentials.with_subject(user_id)
+
+    service = build("gmail", "v1", credentials=delegated_credentials)
+    return service
+
+
+@backoff.on_exception(backoff.expo, TimeoutError, max_tries=10)
+def get_label_list(service: Resource, user_id: str) -> pd.DataFrame:
+    imported_timestamp = datetime.datetime.utcnow().isoformat()
+    label_response = service.users().labels().list(userId=user_id).execute()
+    df_one_label = pd.DataFrame()
+    df_label = pd.DataFrame()
+    for i in range(len(label_response['labels'])):
+        label = label_response['labels'][i]
+        df_one_label['labelId'] = [label['id']]
+        df_one_label['labelName'] = [label['name']]
+        df_one_label['labelType'] = [label['type']]
+        df_label = df_label.append(df_one_label)
+
+    df_label['user_id'] = user_id
+    df_label['imported_timestamp'] = imported_timestamp
+
+    return df_label
+
+
+def iter_link_message_thread(service: Resource, user_id: str) -> Iterable[dict]:
+    response = service.users().messages().list(userId=user_id).execute()
+    if 'messages' in response:
+        yield from response['messages']
+    while ('nextPageToken' in response):
+        page_token = response['nextPageToken']
+        response = service.users().messages().list(userId=user_id, pageToken=page_token).execute()
+        yield from response['messages']
+
+
+def get_link_message_thread(service: Resource, user_id: str) -> pd.DataFrame:
+    df = pd.DataFrame(iter_link_message_thread(service, user_id))
+    return df
+
+
+def write_dataframe_to_file(
```

Comment:
minor: I would maybe suggest to remove the CSV functionality unless we really want to use it. If we then rename it to say `write_dataframe_to_jsonl_file` then it would be quite clear to the caller what file formats it supports.
We have a similar function for regular list of dicts in `data_pipeline/utils/pipeline_file_io.py`, it might be an idea to move it there?

Reply:
It is noted. In case of write to google cloud storage I will keep it here and later I can add it when I deploy all.

Reply:
You could still use JSON when writing to google cloud storage as well.

Reply:
Yes for not to work on different files Just want to keep it here for now. Then I can move it.